phoenix-core/src/it/java/org/apache/phoenix/end2end/index/PartialIndexIT.java

```diff
@@ -1053,4 +1054,44 @@ public void testPartialIndexWithIndexHint() throws Exception {
             assertFalse(rs.next());
         }
     }
+
+    @Test
+    public void testPartialIndexOnTableWithCaseSensitiveColumns() throws Exception {
+        try(Connection conn = DriverManager.getConnection(getUrl());
+            PhoenixStatement stmt = conn.createStatement().unwrap(PhoenixStatement.class)) {
+            String dataTableName = generateUniqueName();
+            String indexName1 = generateUniqueName();
+            String indexName2 = generateUniqueName();
+            String indexName3 = generateUniqueName();
+
+            stmt.execute("CREATE TABLE " + dataTableName
+                    + " (\"hashKeY\" VARCHAR NOT NULL PRIMARY KEY, v1 VARCHAR, \"CoL\" VARCHAR, \"coLUmn3\" VARCHAR)");
+
+            stmt.execute("CREATE INDEX " + indexName1
+                    + " ON " + dataTableName + "(v1) INCLUDE (\"CoL\") WHERE \"coLUmn3\" IS NOT NULL");
+            stmt.execute("CREATE INDEX " + indexName2
+                    + " ON " + dataTableName + "(\"CoL\") INCLUDE (v1) WHERE \"coLUmn3\" IS NOT NULL");
+            stmt.execute("CREATE INDEX " + indexName3
+                    + " ON " + dataTableName + "(\"coLUmn3\") INCLUDE (\"CoL\") WHERE v1 IS NOT NULL");
```

Comment:
nit: Sorry, i should have mentioned this earlier, but can we also add a test for Uncovered Index?

Reply:
@virajjasani I added all the test class parameters to the new test i.e. UNCOVERED, SALTED and LOCAL